include/treelite/c_api.h

```diff
@@ -470,33 +472,74 @@ TREELITE_DLL int TreeliteFreeModel(ModelHandle handle);
  * \{
  */
 
+/*!
+ * \brief Load a configuration for GTIL predictor from a JSON string.
+ * \param config_json a JSON string with the following fields:
+ *   - "nthread" (optional): Number of threads used for initializing DMatrix.
+ *                           Set <= 0 to use all CPU cores.
+ *   - "predict_type" (required): Must be one of the following.
+ *     - "default": Sum over trees and apply post-processing
+ *     - "raw": Sum over trees, but don't apply post-processing; get raw margin scores instead.
+ *     - "leaf_id": Output one (integer) leaf ID per tree.
+ *     - "score_per_tree": Output one or more margin scores per tree.
+ * \param out Parsed configuration
+ * \return 0 for success; -1 for failure
+ */
+TREELITE_DLL int TreeliteGTILParseConfig(const char* config_json, GTILConfigHandle* out);
+
+/*!
+ * \brief Delete a GTIL configuration from memory
+ * \param handle Handle to the GTIL configuration to be deleted
+ * \return 0 for success; -1 for failure
+ */
+TREELITE_DLL int TreeliteGTILDeleteConfig(GTILConfigHandle handle);
+
+/*!
+ * \brief Deprecated. Please use \ref TreeliteGTILGetPredictOutputSizeEx instead.
+ */
+TREELITE_DLL int TreeliteGTILGetPredictOutputSize(ModelHandle model, size_t num_row, size_t* out);
```

Comment:
Default parameters are a C++ feature and are not available for a plain C API.

Context. Treelite exposes a set of functions using a plain C signature, so that they can be called by another language such as Python and Rust. This introduces some limitations:

* Free-standing functions only; no classes or methods.
* No namespaces; use prefixes to avoid name clashes
* No default parameters; all parameter must be explicitly given
* No function overloading.
* Plain types only. C++ objects need to be passed as opaque handles (`void*`).

Reply:
The benefit of exposing a plain C API is that the Treelite Python package does not need Cython. Instead, it uses the `ctypes` module.

Reply:
Thanks a lot for the explanation!